Ednpoints.txt

```diff
@@ -0,0 +1,132 @@
+The same routes for users, chefs and delivery
+POST: /auth/register - registration
+POST: /auth/login - enter to account
+GET: /auth/logout - exit from account
+POST: /auth/forgot-password - request to change password
+POST: /auth/reset-password - password reset
+
+/************/
+
+USER ENDPOINTS:
+
+GET: /users/:userId - get user's info
+PATCH: /users/:userId - update user's info
+DELETE: /users/:userId - delete user's account
+
+GET: /dishes - get a list of dishes
+GET: /dishes/:dishId - get dish information
+GET: /dishes/popular - get popular dishes
+GET: /dishes/random - get some random dish
+
+GET: /chefs - get all chefs
+
+GET: /users/:userId/favourites?type=chef(dish) - get user's favourite dishes or chefs
+	or /users/:userId/favourite-dishes and /users/:userId/favourite-chefs
+POST, DELETE:
+/users/:userId/favourites?type=dishes - add/delete the dish to favourites (passing dishId)
+/users//:userId/favourites?type=chefs - add/delete the chef to favourites (passing chefId)
+
+**user's cart will be stored on the server instead of in local storage
```

Comment:
1. where do you want to store this information, in database?
2. how it will get updates, should we send PATCH requests for every quantity change from UI? or we will have an Update button (have not seen this approach in other marketplaces)?

Reply:
Maybe: 
user:{
   favorites :{
       dishes:[],
       chefs:[],
    }
} 

Dish has button “Favourites” -> update be.

Reply:
@Zaimak-Andrii , as I understand it, we are talking about storing cart items on the server side.

@4m4rel , the team suggested specifically storing data about added dishes on the server, so that when logging in from different devices, the cart would be saved and displayed, as is done, for example, in Rozetka

The shopping cart data is supposed to be stored in the user object

Reply:
I'm pretty sure that if you clear your cache you won't find your added items in cart
usually cart info is stored in browser's `localStorage` as JSON string

there is no 100% correct way, but you need to think of all the possible outcomes for the approach you chose